app/routes.py

```diff
@@ -1,2 +1,253 @@
+        'Authorization': f"Bearer {token}"
+    }
+
+    requests.post(slack_url,json=params,headers=headers)
+    data_response = {
+        "task": {
+            "id": task.task_id,
+            "title": task.title,
+            "description": task.description,
+            "is_complete": True 
+        }
+    }
+
+
+    return make_response(jsonify(data_response),200)
+  
+
+
+@tasks_bp.route("/<task_id>/mark_incomplete", methods = ["PATCH"])
+def handle_incomplete(task_id):
+    task = Task.query.get(task_id)
+
+    if task is None:
+        return make_response("",404)
+    
+    data_response = {
+        "task": {
+            "id": task.task_id,
+            "title": task.title,
+            "description": task.description,
+            "is_complete": False
+        }
+    }
+    db.session.commit()
+    if task.completed_at is None : 
+        return make_response(jsonify(data_response),200)
+    
+    task.completed_at = None
+    return make_response(jsonify(data_response),200)
+
+
+
+
+# Routes for Goal
+
+@goals_bp.route("", methods=["GET","POST"])
+def handle_goals():
+    goal_reponse=[]
+    goals = Goal.query.all()
+  
+    if request.method == "GET":
+    
+        for goal in goals:
+            
+            goal_reponse.append({
+                "id": goal.goal_id,
+                "title":goal.title,
+            })
+        return make_response(jsonify(goal_reponse),200)
+
+    else:
+        request_body = request.get_json()
+        if "title" not in request_body:
+            return make_response({"details": "Invalid data"}, 400)    
+        new_goal = Goal(title = request_body["title"])
+        db.session.add(new_goal)
+        db.session.commit()
+        response_data = {
+        "goal": {
+            'id': new_goal.goal_id,
+            "title":new_goal.title
+
+        }
+    
+    } 
+
+        return make_response(response_data,201)
+
+ 
+
+
+
+@goals_bp.route("/<goal_id>", methods= ["GET","PUT","DELETE"])
+def handle_goal_id(goal_id):
+    goal = Goal.query.get(goal_id)
+    if goal is None:
+        return make_response("",404)
+    if request.method == "GET":
+        if goal is not None:
+            return {
+                "goal":{
+                    "id" : goal.goal_id,
+                    "title": goal.title
+                }
+            }
+    
+    elif request.method == "PUT":
+        data = request.get_json()
+        goal.title= data["title"]
+        db.session.commit()
+
+        data_response= {
+            "goal":{
+                "id": goal.goal_id,
+                "title": goal.title
+            }
+        }
+
+        return make_response(data_response)
+
+    elif request.method == "DELETE":
+        db.session.delete(goal)
+        db.session.commit()
+        delete_id_response= f'Goal {goal_id} "{goal.title}" successfully deleted'
+        return make_response(jsonify({"details":delete_id_response}))
+
+# @goals_bp.route('<goal_id>/tasks', methods=["GET", "POST"])
```

Comment:
To get tests passing for Wave 6 go ahead and implement what you started. Go back and look at the video if you need help with the implementation https://adaacademy.hosted.panopto.com/Panopto/Pages/Viewer.aspx?id=0e99645b-d46b-4e64-b5fe-ad48012e1e3e